File: go-systempromptleakage/secure7.go
package main

import (
	"context"
	"crypto/sha256"
	"encoding/json"
	"fmt"
	"net/http"

	openai "github.com/sashabaranov/go-openai"
)

var client *openai.Client

const systemPrompt = "Internal assistant with access to payroll data."

type PromptInfo struct {
	Hash   string `json:"hash"`
	Length int    `json:"length"`
}

func chatHandler(w http.ResponseWriter, r *http.Request) {
	var body struct {
		Message string `json:"message"`
	}
	json.NewDecoder(r.Body).Decode(&body)
	resp, err := client.CreateChatCompletion(context.Background(), openai.ChatCompletionRequest{
		Model:     openai.GPT4o,
		MaxTokens: 1024,
		Messages: []openai.ChatCompletionMessage{
			{Role: openai.ChatMessageRoleSystem, Content: systemPrompt},
			{Role: openai.ChatMessageRoleUser, Content: body.Message},
		},
	})
	if err != nil {
		http.Error(w, "llm error", http.StatusInternalServerError)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(map[string]string{"answer": resp.Choices[0].Message.Content, "disclaimer": "AI-generated content. Verify independently."})
}

func promptInfoHandler(w http.ResponseWriter, r *http.Request) {
	hash := fmt.Sprintf("%x", sha256.Sum256([]byte(systemPrompt)))[:8]
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(PromptInfo{
		Hash:   hash,
		Length: len(systemPrompt),
	})
}

File: javascript-systempromptleakage/secure7.js
const { OpenAI } = require("openai");
const express = require("express");

const app = express();
const openai = new OpenAI();
const systemPrompt = "Internal assistant configuration.";

app.use(express.json());

app.get("/config", (req, res) => {
    if (!req.user?.isAdmin) {
        return res.status(403).json({ error: "Forbidden" });
    }
    res.json({ prompt: systemPrompt });
});

app.post("/chat", async (req, res) => {
    const completion = await openai.chat.completions.create({
        model: "gpt-4o",
        messages: [
            { role: "system", content: systemPrompt },
            { role: "user", content: req.body.message },
        ],
    });
    res.json({ reply: completion.choices[0].message.content });
});

app.listen(3000);
